packages/expo-updates/android/src/main/java/expo/modules/updates/DisabledUpdatesController.kt

```diff
@@ -26,7 +34,15 @@ class DisabledUpdatesController(
   private val context: Context,
   private val fatalException: Exception?,
   private val isMissingRuntimeVersion: Boolean
-) : IUpdatesController {
+) : IUpdatesController, UpdatesStateChangeEventSender {
+  private var reactNativeHost: WeakReference<ReactNativeHost>? = if (context is ReactApplication) {
+    WeakReference(context.reactNativeHost)
+  } else {
+    null
+  }
+  private val logger = UpdatesLogger(context)
+  private val stateMachine = UpdatesStateMachine(context, this)
```

Comment:
Tough to know. I think the scope of the state machine should be "dumb" in the sense that it should just know how to transition between states and enforce that those transitions are valid. Ideally in the future we might be able to invert the relationship between the state machine and the state machine procedures (make the procedures run upon state change more like how game engines do it rather than the procedures do the state changes), but that's in the future.

Reply:
Discussed this out of band. For now the current approach is fine. In a follow up, we could either pass in a set of allowed states or create a different state machine. Or both: class DisabledFSM extends FSM { ctor() { super(allowedStates=...) }}